proj/hps_accel/gateware/test_sp_mem.py

```diff
@@ -0,0 +1,113 @@
+#!/bin/env python
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from nmigen.sim import Settle
+from nmigen_cfu import TestBase
+
+from .sp_mem import MemoryParameters, SinglePortMemory
+
+
+class SinglePortMemoryTest(TestBase):
+    def create_dut(self):
+        params = MemoryParameters(width=16, depth=32)
+        return SinglePortMemory(params)
+
+    def test(self):
+        X = XX = None
+        A, B, C, D, E, F, G, H, I = range(11, 20)
+
+        # Description
```

Comment:
I had to stare at this test case for quite a while but I think it makes sense to me now

Reply:
Any suggestions about how to format the data table to make it easier to understand would be welcome!

Reply:
Ah no, it wasn't a complaint, I like your use of symbols to represent the arbitrary values, that certainly helped make the table easier to read. I think the part that took time is just taking the large number of input signals at each step and making my brain decipher them and form a good model of what the DUT should be doing so that I can satisfy myself that I understand why the expected values are what they are. That's probably just a fact of life with these kinds of gateware unit tests, I can't think of any way to format them to help with that.